fileUtils.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const lockMap = new Map();
+
+const writeFile = (path, src) => {
+    return new Promise(async (resolve, reject) => {
+        try {
```

Comment:
Why try-catch? Locking can't ever fail.

Reply:
I can't tell what this is intended to catch, since the intent seems to be to use accept/reject, not async.

Reply:
cec8e4c